client_connection/manager.go

```diff
@@ -141,14 +142,16 @@ func ConfigureVpnClientFactory(mysteriumAPIClient server.Client, vpnClientRuntim
 			return nil, err
 		}
 
+		signer := signerFactory(id)
+
 		statsSaver := bytescount.NewSessionStatsSaver(statsKeeper)
-		statsSender := bytescount.NewSessionStatsSender(mysteriumAPIClient, vpnSession.ID, signerFactory(id))
+		statsSender := bytescount.NewSessionStatsSender(mysteriumAPIClient, vpnSession.ID, signer)
 		statsHandler := bytescount.NewCompositeStatsHandler(statsSaver, statsSender)
 
-		authenticator := auth.NewAuthenticatorFake()
+		credentialsProvider := openvpnSession.NewSignedSessionIdCredentialsProvider(vpnSession.ID, signer)
 		vpnMiddlewares := []openvpn.ManagementMiddleware{
 			bytescount.NewMiddleware(statsHandler, 1*time.Minute),
-			auth.NewMiddleware(authenticator),
+			auth.NewMiddleware(credentialsProvider),
```

Comment:
Might be not related to this PR, but it's extremely confusing that `auth.NewMiddleware` can mean two different functions, depending whether you import server or client middleware.

Reply:
We will rethink this packaging later